packages/blade/src/components/Accordion/__tests__/Accordion.native.test.tsx

```diff
@@ -0,0 +1,195 @@
+import { useState } from 'react';
+import { Accordion } from '../Accordion';
+import { AccordionItem } from '../AccordionItem';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';
+import { Button } from '~components/Button';
+
+describe('<Accordion />', () => {
+  // Accordion uses Collapsible which uses animations and requestAnimationFrame which makes RN tests throw warnings
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  it('should render', () => {
+    const { toJSON } = renderWithTheme(
+      <Accordion>
+        <AccordionItem
+          title="How can I setup Route?"
+          description="You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers. You may also check our docs for detailed instructions."
+        />
+        <AccordionItem
+          title="How can I setup QR Codes?"
+          description="Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries."
+        />
+        <AccordionItem
+          title="How can I setup Subscriptions?"
+          description="Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries."
+        />
+      </Accordion>,
+    );
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should expand and close in uncontrolled behavior', async () => {
+    // Suppress an act warning that is likely fixed in recent versions https://github.com/callstack/react-native-testing-library/issues/379
+    const mockConsoleError = jest.spyOn(console, 'error').mockImplementation();
+
+    const onExpandChange = jest.fn();
+    const button1 = 'How can I setup Route?';
+    const description1 =
+      'You can use Razorpay Route from the Dashboard or using APIs to transfer money to customers. You may also check our docs for detailed instructions.';
+    const button2 = 'How can I setup QR Codes?';
+    const description2 =
+      'Just use Razorpay. You may also check our docs for detailed instructions. Please use the search functionality to ask your queries.';
+
+    const { getByText, getAllByRole } = renderWithTheme(
+      <Accordion onExpandChange={onExpandChange}>
+        <AccordionItem title={button1} description={description1} />
+        <AccordionItem title={button2} description={description2} />
+        <AccordionItem title="How can I setup Subscriptions?" description="Just use Razorpay." />
+      </Accordion>,
+    );
+
+    // Clicking first accordion item
+    const trigger1 = getAllByRole('button')[0];
+    expect(trigger1).toHaveAccessibilityState({ expanded: false });
+    expect(getByText(description1)).not.toBeVisible();
+
+    fireEvent.press(trigger1);
+    // wait for animation to finish
+    await waitFor(() => expect(getByText(description1)).toBeVisible());
```

Comment:
we should also assert that description2 should not be visible.

Reply:
`getByText` seems bugged when collapsing back, even if the container has `0` height and `aria-hidden="true"` it still shows the text content as visible. Spent quite a bit of time but couldn't figure out why.

Have added a relevant comment in collapsible tests [here](https://github.com/razorpay/blade/pull/1360/files#diff-777282071cca5e5b49f7bc5a2bba126981123d4d7146480ce9c8b070f1042fa7R126)